Add additional data in GET of Proposals and ProposalResponses

njohner opened the pull request:
In this PR we add some additional data in the GET of proposals and proposal responses. `Proposal`s indeed include important data depending on their status which is not part of their Schema (the meeting it is scheduled in, the excerpt and the decision number). Similarly additional data is required to represent certain `ProposalResponse`s correctly, which is not in the response schema (titles and urls of meeting and successor proposal).

For https://4teamwork.atlassian.net/browse/GEVER-313

## Checklist (Must have)
- [x] Changelog entry
- [x] Documentation updated (notably for API and deployment)
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

elioschmutz commented on opengever/api/proposal.py:
```diff
@@ -1,13 +1,44 @@
+        decision_number = self.context.load_model().get_decision_number()
+        if decision_number:
+            result[u'decision_number'] = decision_number
+        return result
```
Shouldn't the `excerpt`, `meeting`, 'successor_proposals` and the `decision_number` always be included in the response but with `None`? This would make the response more transparent, i.e. b because we know, that there can be a meeting but currently we don't have one...

njohner replied:
I was actually unsure what the behaviour should be here. But fine with me to always send them back. Will amend.

elioschmutz commented on opengever/api/tests/test_proposal.py:
```diff
@@ -73,3 +73,17 @@ def test_submitted_proposal_contains_a_list_of_responses(self, browser):
             responses[0])
+
+    @browsing
+    def test_decided_proposal_contains_meeting_and_decision_number(self, browser):
```
and the excerpt...

njohner commented:
@elioschmutz I've updated according to your comments and added a some more tests (specifically realised that the `decided_proposal` is actually the `SubmittedProposal`, so wrote another test for the proposal on the dossier side...).

As for documentation, we do not have complete responses for all the `portal_type` GETs, so I don't think we need to update them. For proposals we have an entry for `POST` because it's a bit special (https://docs.onegovgever.ch/dev-manual/api/proposals/), but nothing on the GET and IMHO that's ok...
